test/demo/findIntent.inttest.ts

```diff
@@ -0,0 +1,38 @@
+import 'jest';
+
+import {Context, AppIntent, IntentType} from '../../src/client/main';
+
+import * as fdc3Remote from './utils/fdc3RemoteExecution';
+import {testManagerIdentity} from './constants';
+import {setupTeardown} from './utils/common';
+
+setupTeardown();
+
+describe('Resolving intents by context, findIntent', () => {
+    describe('When calling findIntent with an intent type and context type accepted by some directory app', () => {
+        const intent = {
+            type: 'DialCall'
+        };
+        const context = {
+            type: 'fdc3.contact'
+        };
+        test('The promise resolves to an array of all AppIntents which has the specified intent with the specified context or no context at all', async () => {
+            const receivedAppIntents = await findIntent(intent.type, context);
+            expect(receivedAppIntents.apps).toHaveLength(4);
```

Comment:
Wouldn't that mean replicating the whole App data for 6 apps in sample app directory in this test which probably would, again, make it so easily breakable when something in the app-directory data has changed?

I believe that's a risk we run as long as we use a shared app directory data for all tests and not encapsulate them per test? The tests will be breakable regardless when the data is changed in a specific way, would it be reasonable to accept it's the future test writer's responsibility to make sure their change in test data hasn't broken anything else/fix it if needed rather than trying to exhaust all the possible test failures prospectively? 

But I believe your {"name": "test.IntentWithContexts"} approach makes it much less likely to be breakable as that is specific naming of the intent for this test.

Reply:
You wouldn't need to replicate the app data - have a look at how the other tests in `resolveByIntent.inttest.ts` use `extractSortedAppNames`.

You could do `expect(extractSortedAppNames(receivedAppIntents.apps)).toEqual(['test-app-1', ...]);`

Reply:
I have changed this test to check for the names similarly.

While taking a look at other tests tho, I think I might be a little sceptical about the way we use getDirectoryAppsForIntent() to get the expected names for the test. We're basically replicating the functionality of the provider within the test with this method, essentially turns these tests into tautological tests. I'd prefer to hard-code these names in the test to expected values than to use this method.

Reply:
Oh, yeah, I raised an eyebrow at that too. I'll be embarrassed if it turns out I wrote those. Definitely prefer check for specific names